File: src/main/java/com/walking/api/domain/traffic/usecase/AddFavoriteTrafficUseCase.java
package com.walking.api.domain.traffic.usecase;

import com.walking.api.data.entity.member.MemberEntity;
import com.walking.api.data.entity.member.TrafficFavoritesEntity;
import com.walking.api.data.entity.traffic.TrafficEntity;
import com.walking.api.domain.traffic.dto.AddFavoriteTrafficUseCaseIn;
import com.walking.api.repository.dao.traffic.TrafficFavoritesRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Slf4j
@Service
@RequiredArgsConstructor
public class AddFavoriteTrafficUseCase {

	private final TrafficFavoritesRepository trafficFavoritesRepository;

	@Transactional
	public boolean execute(AddFavoriteTrafficUseCaseIn request) {
		TrafficFavoritesEntity entity =
				TrafficFavoritesEntity.builder()
						.memberFk(MemberEntity.builder().id(request.getMemberId()).build())
						.trafficFk(TrafficEntity.builder().id(request.getTrafficId()).build())
						.alias(request.getTrafficAlias())
						.build();

		trafficFavoritesRepository.save(entity);
		return true;
	}
}

File: src/main/kotlin/com/walking/api/traffic/usecase/AddFavoriteTrafficUseCase.kt
package com.walking.api.traffic.usecase

import com.walking.api.data.entity.member.TrafficFavoritesEntity
import com.walking.api.repository.dao.traffic.TrafficFavoritesRepository
import com.walking.api.traffic.dto.AddFavoriteTrafficUseCaseIn
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AddFavoriteTrafficUseCase(private val trafficFavoritesRepository: TrafficFavoritesRepository) {
    @Transactional
    fun execute(request: AddFavoriteTrafficUseCaseIn): Boolean {
        val entity = TrafficFavoritesEntity(
            request.memberId,
            request.trafficId,
            request.trafficAlias
        )
        trafficFavoritesRepository.save(entity)
        return true
    }

    companion object {
        private val log = LoggerFactory.getLogger(
            AddFavoriteTrafficUseCase::class.java
        )
    }
}